src/compiler/scala/tools/nsc/backend/jvm/BTypes.scala

```diff
@@ -847,7 +849,7 @@ abstract class BTypes {
    * a missing info. In order not to crash the compiler unnecessarily, the inliner does not force
    * infos using `get`, but it reports inliner warnings for missing infos that prevent inlining.
    */
-  final case class ClassBType(internalName: InternalName) extends RefBType {
+  final case class ClassBType(internalName: InternalName)(cache: mutable.Map[InternalName, ClassBType]) extends RefBType {
     /**
      * Write-once variable allows initializing a cyclic graph of infos. This is required for
      * nested classes. Example: for the definition `class A { class B }` we have
```

Comment:
Does the introduction of laziness in `nestedClasses` / `innerClasses` allow us to simplify this aspect of the implementation?